thewalrus/quantum.py

```diff
@@ -713,3 +716,68 @@ def is_classical_cov(cov, hbar=2, atol=1e-08):
         if np.all(vals >= 0):
             return True
     return False
+
+
+def gen_single_mode_dist(s, nmax=50, N=1):
+    """Generate the photon number distribution of N identical single mode squeezed states with squeezing parameter s up to cutoff nmax
+    Args:
+        s (float): squeezing paramer
+        nmax (int): Fock cutoff
+        N (float): Number of squeezed state(s).
+    Returns:
+        (array): Photon number distribution
+    """
+    r = 0.5 * N
+    q = 1.0 - np.tanh(s) ** 2
+    N = nmax // 2
+    ps = nbinom.pmf(np.arange(N), p=q, n=r)
+    ps_tot = np.zeros(nmax)
+
+    if nmax % 2 == 0:
+        ps_tot[0::2] = ps
+    else:
+        # This is a bit hacky.
+        ps_tot[0:-1][0::2] = ps
+
+    return ps_tot
+
+
+def gen_multi_mode_dist(s, nmax=50, padding_factor=2):
+    """Generates the total photon number distirbution of len(s) single mode squeezed states with squeezing parameter s[i] up to cutoff
+    nmax.
+    Args:
+        s (array): array of squeezing parameters
+        nmax (int): Fock cutoff
+    Returns:
+        (array): Total photon number distribution

```

Comment:
Is it an integer array?
```suggestion
        (array[int]): total photon number distribution
```